### ISSUE TITLE
Where shall I find the codes for Supplementary Figure 1

### ISSUE THREAD

**Issue by RYY0722** [bowang-lab/scGPT#162]
Hi, is there any update on implementing the Generative attention masking? 
Could you please also provide some explanation in https://github.com/bowang-lab/scGPT/blob/dev-temp/examples/pretrain.py regarding how the mechanism in Fig. S1 is implemented? Thanks a lot!!!

**Comment by subercui** [comment]
Hi, yes, the generative attention masking is implemented in the dev-temp branch. The specific code is here https://github.com/bowang-lab/scGPT/blob/dev-temp/scgpt/model/flash_layers.py#L19. It takes in two parts of input pcpt_total_embs, gen_total_embs. The first relates to the genes with expression value input, and the second is the genes whose expression values are to be predicted. We also recently worked on a faster version that uses the triton flash attention, and we are testing it and will release soon

**Comment by RYY0722** [comment]
Thank you!
